### ISSUE TITLE
0.60 Winget auto-release failed

### ISSUE THREAD

**Issue by jntrnr** [nushell/nushell#4894]
### Describe the bug

```
Submitting pull request for manifest...
[92](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:92)

[93](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:93)
Octokit.NotFoundException: Not Found
[94](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:94)
   at Octokit.Connection.HandleErrors(IResponse response)
[95](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:95)
   at Octokit.Connection.RunRequest(IRequest request, CancellationToken cancellationToken)
[96](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:96)
   at Octokit.Connection.Run[T](IRequest request, CancellationToken cancellationToken)
[97](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:97)
   at Octokit.ApiConnection.Post[T](Uri uri, Object data, String accepts, String contentType, CancellationToken cancellationToken)
[98](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:98)
   at Microsoft.WingetCreateCore.Common.GitHub.<>c__DisplayClass19_0.<<SubmitPRAsync>b__1>d.MoveNext()
[99](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:99)
--- End of stack trace from previous location ---
[100](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:100)
   at Polly.AsyncPolicy.<>c__DisplayClass40_0.<<ImplementationAsync>b__0>d.MoveNext()
[101](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:101)
--- End of stack trace from previous location ---
[102](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:102)
   at Polly.Retry.AsyncRetryEngine.ImplementationAsync[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Func`5 onRetryAsync, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider, Boolean continueOnCapturedContext)
[103](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:103)
   at Polly.AsyncPolicy.ExecuteAsync(Func`3 action, Context context, CancellationToken cancellationToken, Boolean continueOnCapturedContext)
[104](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:104)
   at Microsoft.WingetCreateCore.Common.GitHub.SubmitPRAsync(String packageId, String version, Dictionary`2 contents, Boolean submitToFork)
[105](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:105)
   at Microsoft.WingetCreateCore.Common.GitHub.SubmitPRAsync(String packageId, String version, Dictionary`2 contents, Boolean submitToFork)
[106](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:106)
   at Microsoft.WingetCreateCLI.Commands.BaseCommand.GitHubSubmitManifests(Manifests manifests)
[107](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:107)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.ExecuteManifestUpdate(List`1 latestManifestContent, CommandExecutedEvent commandEvent)
[108](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:108)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.Execute()
[109](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:109)
   at Microsoft.WingetCreateCLI.Program.Main(String[] args)
[110](https://github.com/nushell/nushell/runs/5651207700?check_suite_focus=true#step:2:110)
{"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/git#create-a-reference"}
```

### How to reproduce

Automatically happens on a release

### Expected behavior

This should successfully send information to winget so they can publish an update

**Comment by jntrnr** [comment]
@TechWatching @officialesco - apologies for the ping. just noticed that the winget script failed again for today's release. It's a different error than https://github.com/nushell/nushell/issues/4368

**Comment by TechWatching** [comment]
@jntrnr It could be linked to [this issue](https://github.com/microsoft/winget-create/issues/233). A workaround is suggested [here](https://github.com/microsoft/winget-create/issues/233#issuecomment-1061241489). Could you try it by updating your fork of winget packages repository if it's your PAT that is used (sorry I don't remember if it's yours or someone else)?

**Comment by denelon** [comment]
We also recently ran into [this issue](https://github.com/microsoft/winget-create/issues/241). We also just released 1.0 today.

**Comment by TechWatching** [comment]
Okay. So if I understand correctly @denelon , if we re-run our pipeline it should take the 1.0.3 version of wingetcreat that solves the issue you mentioned. Then maybe @jntrnr you can update your fork of winget-pkg, and re-run the pipeline?

**Comment by denelon** [comment]
Yes, I'm hoping that it will work now.

**Comment by fdncred** [comment]
I'm guessing it's my fork of winget-pkgs. I just updated it `This branch is up to date with microsoft/winget-pkgs:master.`

**Comment by fdncred** [comment]
@TechWatching I tried re-running it after updating my winget-pkg repo. It fails with this error now. It looks like the parameters are wrong or something like that.


```
Run iwr https://aka.ms/wingetcreate/latest -OutFile wingetcreate.exe
[2](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:2)
  iwr https://aka.ms/wingetcreate/latest -OutFile wingetcreate.exe
[3](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:3)
  $github = Get-Content 'D:\a\_temp\_github_workflow\event.json' | ConvertFrom-Json
[4](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:4)
  $installerUrl = $github.release.assets | Where-Object -Property name -match 'windows.msi' | Select -ExpandProperty browser_download_url -First 1
[5](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:5)
  .\wingetcreate.exe update Nushell.Nushell -s -v $github.release.tag_name -u $installerUrl -t  ***
[6](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:6)
  shell: C:\Program Files\PowerShell\[7](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:7)\pwsh.EXE -command ". '{0}'"
7
Telemetry Settings
[8](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:8)
------------------
[9](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:9)
The Windows Package Manager Manifest Creator collects usage data in order to improve your experience.
[10](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:10)
The data is anonymous and collected only by Microsoft.
[11](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:11)
By default, telemetry is enabled but can be disabled by running `wingetcreate settings` and editing your settings file.
[12](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:12)

[13](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:13)
Windows Package Manager Manifest Creator v1.0.3.0
[14](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:14)

[15](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:15)
Copyright (c) Microsoft Corporation. All rights reserved.
[16](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:16)

[17](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:17)
The manifest creation command line utility generates manifest for submitting
[18](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:18)
apps to the Windows Package Manager repo.
[19](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:19)

[20](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:20)
USAGE:
[21](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:21)
Search for an existing manifest and update the installer url and version:
[22](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:22)
  wingetcreate.exe update --urls <InstallerUrl1> <InstallerUrl2> --version <Version>
[23](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:23)
  <PackageIdentifier>
[24](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:24)
Save and publish updated manifest:
[25](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:25)
  wingetcreate.exe update --out <OutputDirectory> --token <GitHubPersonalAccessToken> --version
[26](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:26)
  <Version> <PackageIdentifier>
[27](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:27)
Override the architecture of an installer:
[28](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:28)
  wingetcreate.exe update --urls <InstallerUrl1>|<InstallerArchitecture> --version <Version>
[29](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:29)
  <PackageIdentifier>
[30](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:30)

[31](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:31)
  -v, --version                 Version to be used when updating the package version field.
[32](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:32)
  -o, --out                     Output directory where the newly created manifests will be saved
[33](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:33)
                                locally
[34](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:34)
  -s, --submit                  Boolean value for submitting to the Windows Package Manager repo. If
[35](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:35)
                                true, updated manifest will be submitted directly using the provided
[36](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:36)
                                GitHub Token. Default is false.
[37](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:37)
  -i, --interactive             Boolean value for making the update command interactive. If true,
[38](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:38)
                                the tool will prompt the user for input. Default is false.
[39](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:39)
  -t, --token                   GitHub personal access token used for direct submission to the
[40](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:40)
                                Windows Package Manager repo. If no token is provided, tool will
[41](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:41)
                                prompt for GitHub login credentials.
[42](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:42)
  -u, --urls                    Installer Url(s) used to extract relevant metadata for generating a
[43](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:43)
                                manifest
[44](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:44)
  PackageIdentifier (pos. 0)    Required. Package identifier used to lookup the existing manifest on
[45](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:45)
                                the Windows Package Manager repo. Id is case-sensitive.
[46](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:46)

[47](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:47)
More help can be found at: https://aka.ms/winget-create
[48](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:48)
Privacy statement: https://aka.ms/privacy
[49](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:49)

[50](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:50)
 Option 'v, version' has no value.
[51](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:51)
 Option 'u, urls' has no value.
[52](https://github.com/nushell/nushell/runs/5654055755?check_suite_focus=true#step:2:52)
Error: Process completed with exit code 1.
```

**Comment by denelon** [comment]
I've asked someone from the Windows Package Manager Manifest Creator team to take a look at this. It will probably be a few hours before they get in and have a chance.

**Comment by TechWatching** [comment]
@fdncred From what I see you tried manually launching another run. But that can't work as the workflow get the value of `$installerUrl` from the release being published that triggered it. I was not specific enough but when I said we could try to "re-run" the workflow I was saying re-run the workflow instance that failed when release 0.60.0 was published so this [one](https://github.com/nushell/nushell/actions/runs/2024870490). You should have a button "Re-run workflow" or "Re-run all failed jobs" I guess.

**Comment by fdncred** [comment]
Thanks for the support and tip. Sorry I messed it up. I see what you're saying now. For now, someone has already posted a 0.60 in winget so I guess we'll wait for the next release in a few weeks.

**Comment by OfficialEsco** [comment]
> Thanks for the support and tip. Sorry I messed it up. I see what you're saying now. For now, someone has already posted a 0.60 in winget so I guess we'll wait for the next release in a few weeks.

Oh, our community made bot also tried to update it, @vedantmgoyal2009 you should probably exclude this since they can maintain the package

**Comment by TechWatching** [comment]
> Thanks for the support and tip. Sorry I messed it up. I see what you're saying now. For now, someone has already posted a 0.60 in winget so I guess we'll wait for the next release in a few weeks.

No problem. As you say, we will see the next release if everything works fine again.

**Comment by fdncred** [comment]
Let's close this since 0.60 is published and open a new issue on 0.61 if needed.
